File: 2_Ruby_Examples/2_Shapes_LSP/Main.java
public class Main {
	
	public static void main(String[] args) {
		Rectangle rect = new Rectangle(0,0,10,20);
		Square sqr = new Square(0,0,10);
		scale(rect,2,3);
		scale(sqr,2,3);
	}
	
	private void scale(Rectangle rect, int horizontalScaleFactor, int verticalScaleFactor) {
		rect.setSideLength(rect.getHorizontalLength() * horizontalFactor, rect.getVerticalLength() * verticalFactor);
	}
	
}

File: 2_Ruby_Examples/2_Shapes_LSP/main.rb
require 'square'

def scale rectangle, horizontal_scale_factor, vertical_scale_factor
	rectangle.set_side_lengths(rectangle.horizontal_length * horizontal_scale_factor, rectangle.vertical_length * vertical_scale_factor)
end

rectangle = Rectangle.new(0,0,10,20)
square = Square.new(0,0,10)
scale(rectangle,2,3)
scale(square,2,3)
